src/main/java/seedu/socket/logic/commands/EditProjectCommand.java

```diff
@@ -0,0 +1,220 @@
+package seedu.socket.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.socket.logic.parser.CliSyntax.PREFIX_DEADLINE;
+import static seedu.socket.logic.parser.CliSyntax.PREFIX_MEETING;
+import static seedu.socket.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.socket.logic.parser.CliSyntax.PREFIX_REPO_HOST;
+import static seedu.socket.logic.parser.CliSyntax.PREFIX_REPO_NAME;
+import static seedu.socket.model.Model.PREDICATE_SHOW_ALL_PROJECTS;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.socket.commons.core.Messages;
+import seedu.socket.commons.core.index.Index;
+import seedu.socket.commons.util.CollectionUtil;
+import seedu.socket.logic.commands.exceptions.CommandException;
+import seedu.socket.model.Model;
+import seedu.socket.model.person.Person;
+import seedu.socket.model.project.Project;
+import seedu.socket.model.project.ProjectDeadline;
+import seedu.socket.model.project.ProjectMeeting;
+import seedu.socket.model.project.ProjectName;
+import seedu.socket.model.project.ProjectRepoHost;
+import seedu.socket.model.project.ProjectRepoName;
+
+
+
+/**
+ * Edits the details of an existing person in SOCket.
+ */
+public class EditProjectCommand extends Command {
+
+    public static final String COMMAND_WORD = "editpj";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of this project identified "
+            + "by the index number used in the displayed project list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "NAME] "
+            + "[" + PREFIX_REPO_HOST + "REPO HOST] "
+            + "[" + PREFIX_REPO_NAME + "REPO NAME] "
+            + "[" + PREFIX_DEADLINE + "DEADLINE] "
+            + "[" + PREFIX_MEETING + "MEETING]...\n "
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_NAME + "Socket2.0 "
+            + PREFIX_DEADLINE + "02/01/2023-2359";
+
+    public static final String MESSAGE_EDIT_PROJECT_SUCCESS = "Edited Project: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_PROJECT = "This person already exists in SOCket.";
```

Comment:
Seems like editing the string was missed? Perhaps it should be 
`"This project already exists in SOCket."` 
instead?